### DISCUSSION TITLE
Deserialize "epoch" in milliseconds

### DISCUSSION THREAD

**Discussion by mbergkvist** [time-rs/time#410] [Question] [answered]
I'm deserializing a JSON data where one of the field is a unix timestamp in milliseconds. With chrono I'm using `#[serde(deserialize_with = "chrono::serde::ts_milliseconds::deserialize")]` but I'm now looking into replacing `chrono` with `time` but I can't seem to get the deserializer for `time::OffsetDateTime` to work. Any hints?

**Comment by jhpratt** [comment, marked as answer]
What code do you have? I presume you're using a custom deserializer, as there's not one in this library with the exact behavior you're looking for (I imagine you already noticed this).

**Reply by mbergkvist** [reply]
I was looking for a deserializer and couldn't find one, but I was hoping that it was my search skill that was lacking and not the a ready-to-use-deserializer in `time` ;-)

Any pointer to start looking if I'm to implement it myself?

**Reply by jhpratt** [reply]
Probably the best place to look is the [source for the timestamp](https://github.com/time-rs/time/blob/fa5c0a083c29432b6ded8fa49df2ac1c67c94612/src/serde/timestamp.rs) serde implementation. Only slight modifications should be necessary to have it use milliseconds.

**Reply by mbergkvist** [reply]
Thanks. I'll look into it.

**Comment by mbergkvist** [comment]
I ended up with a deserializer looking like this

```rust
pub(super) fn deserialize<'de, D>(deserializer: D) -> Result<OffsetDateTime, D::Error>
where
    D: Deserializer<'de>,
{
    let ms = i64::deserialize(deserializer)?;
    let sec = ms / 1000;
    OffsetDateTime::from_unix_timestamp(sec)
        .map_err(|err| de::Error::invalid_value(de::Unexpected::Signed(sec), &err))
}
```
